transitmatters/t-performance-dash
# add csv download link to speed, prediction accuracy, service, ridership

## Motivation

Commonly used widgets need a link to download a .csv file so people can use our data in research without needing to reach out to us [#906](https://github.com/transitmatters/t-performance-dash/issues/906)

## Changes
![Screenshot 2024-01-11 at 12 32 58 PM](https://github.com/transitmatters/t-performance-dash/assets/47030097/a05de15e-3ae9-4e1e-b416-d9a68f7f26f9)
<img width="1224" alt="Screenshot 2024-01-13 at 5 22 18 PM" src="https://github.com/transitmatters/t-performance-dash/assets/47030097/e85c60f2-ec3c-40cf-9cb5-c7a0849ec90c">
<img width="610" alt="Screenshot 2024-01-13 at 5 22 36 PM" src="https://github.com/transitmatters/t-performance-dash/assets/47030097/c45dc0d4-06cc-4e26-8ef4-7f0021ad10f1">
<img width="598" alt="Screenshot 2024-01-13 at 5 22 57 PM" src="https://github.com/transitmatters/t-performance-dash/assets/47030097/49420bea-587a-43ed-a2a6-d03e87186b3e">
<img width="510" alt="Screenshot 2024-01-13 at 5 23 10 PM" src="https://github.com/transitmatters/t-performance-dash/assets/47030097/12c2791b-c7e1-4f56-97a7-4ab368fc5f15">
<img width="179" alt="Screenshot 2024-01-13 at 5 23 24 PM" src="https://github.com/transitmatters/t-performance-dash/assets/47030097/6ef3a0cd-d9c1-437e-8dc9-b8936561ba14">
<img width="1217" alt="Screenshot 2024-01-13 at 5 23 37 PM" src="https://github.com/transitmatters/t-performance-dash/assets/47030097/fd7aa8ba-25fc-44bf-a33b-bc4ec49fab73">
<img width="1223" alt="Screenshot 2024-01-13 at 5 23 47 PM" src="https://github.com/transitmatters/t-performance-dash/assets/47030097/c246756a-5dda-40d1-be13-ff69c9fe3f69">
<img width="1222" alt="Screenshot 2024-01-13 at 5 23 59 PM" src="https://github.com/transitmatters/t-performance-dash/assets/47030097/ac6361b6-a1df-47aa-b5c2-f6bb69d03e1b">
<img width="1218" alt="Screenshot 2024-01-13 at 5 24 16 PM" src="https://github.com/transitmatters/t-performance-dash/assets/47030097/eccfdae6-b7de-4064-9647-e6e4a1501602">
<img width="861" alt="Screenshot 2024-01-13 at 5 24 26 PM" src="https://github.com/transitmatters/t-performance-dash/assets/47030097/b2c7a866-089b-4f38-8069-959fbe11c979">

## Testing Instructions
1. Load the app locally
2. Widgets on the Line Overview page should have download links (Speed, Service, Ridership). Arrival Predictions should have a download button as well. 
3. Test by adjusting time frame on the top right, click download button for widget, ensure data is within time range
